src/storage/CookiesStorage.js

```diff
@@ -67,18 +67,46 @@ class CookiesStorage {
     if (document.cookie === oldCookies) {
       return null;
     }
-    
+
     // Since the cookie was updated with a dummy value, remove it by setting
     // expiry to a date in the past.
-    const pastDate = "Thu, 01 Jan 1970 00:00:00 GMT";
-    document.cookie = 
+    const pastDate = 'Thu, 01 Jan 1970 00:00:00 GMT';
+    document.cookie =
       `registrable_domain=;domain=${domain};expires=${pastDate}`;
 
     return domain;
   }
 
   /** @override */
-  save(key, value, secondsToLive) {}
+  save(key, value, secondsToLive =
+    /** @type {?number} */ (this.settings_['expires'])) {
+    const valString = JSON.stringify(value);
+    let newCookie = `${this.settings_['prefix']}${key}=${valString}`;
+    newCookie = this.addSettings(newCookie, secondsToLive);
+    document.cookie = newCookie;
+  }
+
+  /**
+   * Finds the registrable domain of current URL and sets it as the default
+   *    domain for cookies
+   * @param {string} cookie the cookie to add settings to.
+   * @param {?number} secondsToLive number of seconds cookie will be in storage.
+   * @return {string} the cookie with all added settings.
+   * @private
+   */
+  addSettings(cookie, secondsToLive) {
+    cookie = `${cookie}; domain=${this.settings_['domain']}`;
+
+    const expiry = new Date();
+    expiry.setMilliseconds(expiry.getMilliseconds() + (secondsToLive + 1) *
+      1000);
```

Comment:
Why are you adding 1 here? It's [documented](https://github.com/googleinterns/measurement-library/tree/5b17e8fbe7341eadd67fc16964f8b2f0ffb5e5a1#set-command) that if secondsToLive is 0, then the value should not be saved, if that is your concern. I thought it was written [here](https://github.com/googleinterns/measurement-library/blob/5b17e8fbe7341eadd67fc16964f8b2f0ffb5e5a1/src/storage/StorageInterface.js) but apparently not, might be worth adding that in.

Reply:
Actually, NUMBER.POSITIVE_INFINITY may present an edge case for you, since cookies can't be set to never expire. Maybe add a check and store it for several years?